haskell-src/Concordium/Types/ProtocolLevelTokens/CBOR.hs

```diff
@@ -1066,16 +1093,58 @@ data TokenEvent
       AddDenyListEvent !CborTokenHolder
     | -- | An account was removed from the deny list.
       RemoveDenyListEvent !CborTokenHolder
+    | -- | Whether the execution of token operations has been paused.
+      Pause
+    | -- | Whether the execution of token operations has been unpaused.
+      Unpause
     deriving (Eq, Show)
 
--- | CBOR-encode the details for the list update events in the form:
---  > {"target": <TokenHolder>}
-encodeTargetDetails :: CborTokenHolder -> TokenEventDetails
-encodeTargetDetails target =
-    TokenEventDetails . BSS.toShort . CBOR.toStrictByteString $
-        encodeMapLen 1
-            <> encodeString "target"
-            <> encodeCborTokenHolder target
+-- | CBOR-encode the details for the token events in the form:
+--  > {"label": <CborEncodedValue>}
+encodeTokenEventDetails :: Maybe Text.Text -> (a -> Encoding) -> a -> TokenEventDetails
+encodeTokenEventDetails mbLabel encoder x = case mbLabel of
+    Nothing -> TokenEventDetails . BSS.toShort . CBOR.toStrictByteString $ encodeMapLen 0
+    Just label ->
+        TokenEventDetails . BSS.toShort . CBOR.toStrictByteString $
+            encodeMapLen 1
+                <> encodeString label
+                <> encoder x
+
+-- | Decoder for the event details of token events.
+decodeTokenEventDetails ::
+    -- | The (single) field name of the event details
+    Text.Text ->
+    -- | The decoder for the field
+    (forall s. Decoder s a) ->
+    -- | The input bytestring
+    LBS.ByteString ->
+    Either String a
+decodeTokenEventDetails label innerDecoder detailsLBS =
+    case CBOR.deserialiseFromBytes (decodeWith innerDecoder) detailsLBS of
+        Left e -> Left $ "token-event: failed to decode event details: " ++ show e
+        Right ("", target) -> Right target
+        Right (remaining, _) ->
+            Left $
+                "token-event: "
+                    ++ show (LBS.length remaining)
+                    ++ " bytes remaining after parsing event details"
+  where
+    decodeWith decoder = do
+        maybeMapLen <- decodeMapLenOrIndef
+        forM_ maybeMapLen $ \mapLen ->
+            unless (mapLen == 1) $
+                fail $
+                    "token-event: expected a map of size 1, but saw " ++ show mapLen
+        label' <- decodeString
+        unless (label' == label) $
+            fail $
+                "token-event: expected \"target\" key, but saw "
+                    ++ show label'
+        x <- decoder
+        when (isNothing maybeMapLen) $ do
+            isEnd <- decodeBreakOr
+            unless isEnd $ fail "token-event: expected end of map"
+        return x
```

Comment:
Does this not still need an update? from what I can see, this still only handles the list update events with a map with a single entry "target"? Or am I missing something?